File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* middleNode(struct ListNode* head){
    int len = 0;
    struct ListNode* tmp = head;
    
    if(head == NULL)
        return NULL;
    while(tmp->next != NULL)
    {
        ++len;
        tmp = tmp->next;
    }
    ++len;
    len = len/2;
    while (len > 0)
    {
        head = head->next;
        len--;
    }
    return head;
}

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode tmp = head;
        int len = 0;
        while(tmp != null){
            len++;
            tmp = tmp.next;
        }
        len /= 2;
        tmp = head;
        while(len != 0){
            tmp = tmp.next;
            len--;
        }
        return tmp;
    }
}